frontend/packages/console-dynamic-plugin-sdk/src/utils/dynamic-module-parser.ts

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable no-console */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as glob from 'glob';
+import * as _ from 'lodash';
+import * as ts from 'typescript';
+
+const defaultCompilerOptions: ts.CompilerOptions = {
+  target: ts.ScriptTarget.ES2020,
+  module: ts.ModuleKind.ESNext,
+  moduleResolution: ts.ModuleResolutionKind.NodeJs,
+  allowJs: true,
+  strict: false,
+  esModuleInterop: true,
+  skipLibCheck: true,
+  noEmit: true,
+};
+
+/**
+ * Maps each index module export to a dynamic module request (import specifier).
+ */
+export type DynamicModuleMap = { [exportName: string]: string };
+
+/**
+ * Map all exports of the given index module to their corresponding dynamic modules.
+ *
+ * Example: `@patternfly/react-core` package provides ESModules index at `dist/esm/index.js`
+ * which exports Alert component related code & types via `dist/esm/components/Alert/index.js`
+ * module.
+ *
+ * Given the example above, this function should return a mapping like so:
+ * ```js
+ * {
+ *   Alert: 'dist/dynamic/components/Alert',
+ *   AlertProps: 'dist/dynamic/components/Alert',
+ *   AlertContext: 'dist/dynamic/components/Alert',
+ *   // ...
+ * }
+ * ```
+ *
+ * The above mapping can be used when generating import statements like so:
+ * ```ts
+ * import { Alert } from '@patternfly/react-core/dist/dynamic/components/Alert';
+ * ```
+ *
+ * It may happen that the same export is provided by multiple dynamic modules; in such case,
+ * the resolution favors non-deprecated modules with most specific file paths, for example
+ * `dist/dynamic/components/Wizard/hooks` is favored over `dist/dynamic/components/Wizard`.
+ *
+ * If the referenced index module does not exist, an empty object is returned.
+ */
+export const getDynamicModuleMap = (
+  basePath: string,
+  indexModule = 'dist/esm/index.js',
+  resolutionField = 'module',
+  tsCompilerOptions = defaultCompilerOptions,
+): DynamicModuleMap => {
+  if (!path.isAbsolute(basePath)) {
+    throw new Error('Package base path must be absolute');
+  }
+
+  const indexModulePath = path.resolve(basePath, indexModule);
+
+  if (!fs.existsSync(indexModulePath)) {
+    return {};
+  }
+
+  const dynamicModulePathToPkgDir = glob
+    .sync(`${basePath}/dist/dynamic/**/package.json`)
+    .reduce<Record<string, string>>((acc, pkgFile) => {
+      // eslint-disable-next-line
```

Comment:
nitpick: better to only disable the eslint rule we need

Reply:
@spadgett The `require` call (line below) actually triggers multiple ESLint errors:
```
  72:19  error  Calls to require() should use string literals   import/no-dynamic-require
  72:19  error  Unexpected require()                            global-require
  72:19  error  A `require()` style import is forbidden         @typescript-eslint/no-require-imports
  72:19  error  Require statement not part of import statement  @typescript-eslint/no-var-requires
```
so I didn't specify the rule names explicitly. Such `require` calls always tend to trigger a lot of ESLint errors.

Is it OK to leave it like so for simplicity?

We do this also in some other parts of the codebase, e.g. `packages/console-dynamic-plugin-sdk/src/runtime/plugin-manifest.ts`

Reply:
+1 leaving as is